runtime/src/main/java/org/corfudb/protocols/wireprotocol/ExceptionMsg.java

```diff
@@ -0,0 +1,28 @@
+package org.corfudb.protocols.wireprotocol;
+
+import org.corfudb.util.serializer.Serializers;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+
+/** This message wraps exceptions that are encountered on the server.
+ *
+ */
+public class ExceptionMsg implements ICorfuPayload<ExceptionMsg> {
+
+    @Getter
+    final Throwable throwable;
+
+    public ExceptionMsg(Throwable t) {
+        throwable = t;
+    }
+
+    public ExceptionMsg(ByteBuf b) {
+        throwable = (Throwable) Serializers.JAVA.deserialize(b, null);
+    }
+
+    @Override
+    public void doSerialize(ByteBuf buf) {
+        Serializers.JAVA.serialize(throwable, buf);
```

Comment:
Assuming we have the class, it should work (vis kryo). I'll catch the exception nevertheless.